test/rest-router.test.js

```diff
@@ -0,0 +1,60 @@
+var assert = require('assert');
+var expect = require('chai').expect;
+var RestRouter = require('../lib/rest-router');
+
+describe('Custom Router', function() {
+  var restRouter;
+  var route = { verb: 'get', fullPath: '/Planets' };
+  var handler = function(req, res, next) { res.handled = true; };
+  var req = {
+    method: 'get',
+    url: '/Planets',
+  };
+  var res = {};
```

Comment:
Mocking out req/res objects is a bad idea, Express maintainers are actively discouraging people from doing that.

Please create a regular express app to test the router. Also keep in mind that you need to write tests for both values of `options.caseSensitive`.

```js
var app;
beforeEach(function() {
  app = expres();
});

it('invokes the handler function for matching verb+path', function(done) { 
  var router = new RestRouter(); // optionally set options here
  app.use(router);
  router.get('/Planets', function(req, res, next) { res.end('hello'); });
  executeRouter(app, 'get', '/Planets', function(err, response, body) {
    expect(body).to.equal('hello');
  });
});

function executeRouter(app, verb, urlPath, callback) {
  var server = http.createServer(function(req, res) {
    app.handle(req, res, callback);
  });

  supertest(server)[verb](urlPath).end(callback);
}
```

This is just a mockup based on https://github.com/strongloop/loopback/blob/9a33602458712afecedd25062f02bd8b5c4e01a5/test/app.test.js#L959-L974, it needs a bit of polishing. I think ultimately we would like to arrive at something like this:

```js
beforeEach(function(done) {
  // setup the app
  // setup var request = supertest(app or server)
});

it('does something', function(done) {
  // setup the router
  // register handlers (routes)
  request.get('/path').expect(200).expect('expected-body').end(done);
});
```